spyder/plugins/completion/providers/languageserver/providers/document.py

```diff
@@ -309,7 +309,7 @@ def document_did_close(self, params):
                 if id(codeeditor) == id(editor):
                     idx = i
                     break
-            if idx > 0:
+            if idx >= 0:
```

Comment:
Why is it necessary to the equal sign here? Just asking because I don't understand it.

Reply:
I think this was an error in the code. If no CodeEdotor are found, the index should be -1, so you can check with index >=0 (and not >0 which leads to a leak)